### DISCUSSION TITLE
Changing process variables after deploy

### DISCUSSION THREAD

**Discussion by Anton1922** [camunda-community-hub/zeebe-client-node-js#335] [General]
Hi!
I use
**camunda/zeebe 8.2.11
zeebe-node 8.0.1**

I'm new to using zeebe and zeebe-node. I read the documentation but didn't find a solutions for my questions. 
1. Can I set variables, that will be available to the whole process, using method `deployProcess` ?
2. Can I change the values of the variables for the process after deploy? I found the `setVariables` method, but as I understand it, this method sets variables for a specific process instance by its id. Or am I wrong?

I will be glad for help!

Best regards,
Anton Fedorov

**Comment by jwulf** [comment]
Hi @Anton1922 

`deployProcess` is to deploy a process model. Variables are state in a process instance. 

`createProcessInstance` is how you create an instance of a model (a "process instance"), and you can set initial variables in that call.
